### PR TITLE
Adds Watches on NADs

### PR DESCRIPTION
**Pull request by stuggi**
Adds to watch NADs so that the controller reconciles when the configured NAD changes.

### CODE REVIEW

File: controllers/keystoneapi_controller.go [hunk 1]
```diff
@@ -297,6 +300,18 @@ func (r *KeystoneAPIReconciler) SetupWithManager(ctx context.Context, mgr ctrl.M
 		return err
 	}
 
+	// index networkAttachmentsField
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &keystonev1.KeystoneAPI{}, networkAttachmentsField, func(rawObj client.Object) []string {
+		// Extract the secret name from the spec, if one is provided
```
Comment (abays) [inline controllers/keystoneapi_controller.go:305]: It isn't a `Secret` in the case, right?  Maybe we should change the comment.

Comment (stuggi) [reply controllers/keystoneapi_controller.go]: yes, updated

**Review by abays — APPROVED** [review]
/lgtm

**Comment by stuggi** [comment]
closing in favor of https://github.com/openstack-k8s-operators/keystone-operator/pull/459

**Pull request closed without merging**